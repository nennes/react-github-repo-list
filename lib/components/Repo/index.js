/* @flow */

import React from "react";
import Name from "../../elements/Name";
import Title from "../../elements/Title";
import ReadMe from "../../elements/ReadMe";
import LastUpdated from "../../elements/LastUpdated";

const Repo = ({
  name,
  title,
  url,
  markdown,
  timestamp
}: {
  name: string,
  title: string,
  url: string,
  markdown: string,
  timestamp: string
}) => {
  return (
    <div className="repo">
      <p>
        <Name name={name} url={url} />
      </p>
      <p>
        <Title title={title} />
      </p>
      <p>
        <ReadMe markdown={markdown} />
        <LastUpdated timestamp={timestamp} />
      </p>
    </div>
  );
};

export default Repo;
