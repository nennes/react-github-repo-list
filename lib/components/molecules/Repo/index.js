/* @flow */

import React from "react";
import { IntlProvider } from "react-intl";
import { Repo as RepoType } from "../../../types";
import Name from "../../atoms/Name";
import Title from "../../atoms/Title";
import ReadMe from "../../atoms/ReadMe";
import LastUpdated from "../../atoms/LastUpdated";

const Repo = ({ name, title, url, markdown, timestamp }: RepoType) => {
  return (
    <IntlProvider locale="en">
      <div className="repo">
        <div>
          <Name name={name} url={url} />
        </div>
        <div>
          <Title title={title} />
        </div>
        <div>
          <ReadMe markdown={markdown} />
          <LastUpdated timestamp={timestamp} />
        </div>
      </div>
    </IntlProvider>
  );
};

export default Repo;
