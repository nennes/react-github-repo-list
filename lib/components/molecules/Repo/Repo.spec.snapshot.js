/* global test, expect */

import React from "react";
import Repo from "./index";
import renderer from "react-test-renderer";

test("Repo component renders correctly", () => {
  const props = {
    markdown: "# Some markdown",
    name: "repository-name",
    timestamp: "2015-12-13T13:36:22Z",
    title: "Repository title",
    url: "http://www.test.com"
  };
  const component = renderer.create(<Repo {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
