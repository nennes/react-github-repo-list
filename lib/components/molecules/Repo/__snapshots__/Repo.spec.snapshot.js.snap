// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo component renders correctly 1`] = `
<div
  className="repo"
>
  <div>
    <div
      className="repo__name"
    >
      <a
        href="http://www.test.com"
        target="_blank"
      >
        <span
          title="repository-name"
        >
          repository-name
        </span>
      </a>
    </div>
  </div>
  <div>
    <span
      className="repo__title"
    >
      <h3>
        Repository title
      </h3>
    </span>
  </div>
  <div>
    <div
      className="repo__read-me"
    >
      <button
        onClick={[Function]}
      >
        ReadMe
      </button>
    </div>
    <div
      className="repo__last-updated"
    >
      Last Updated: 
      <span>
        2 years ago
      </span>
    </div>
  </div>
</div>
`;
